File: test-dagl-mixed/analyze_fn.py
def handler(event, context):
    """Python version of Analyze — same interface as the Node.js one."""
    tokens = event.get("tokens", [])
    freq = {}
    for word in tokens:
        freq[word] = freq.get(word, 0) + 1
    sorted_words = sorted(freq.items(), key=lambda x: -x[1])
    return {
        "totalWords": len(tokens),
        "uniqueWords": len(sorted_words),
        "topWords": [{"word": w, "freq": c} for w, c in sorted_words[:5]],
    }

File: test-dagl-nodejs/analyze_fn.js
exports.handler = async (input, context) => {
  const tokens = input.tokens || [];
  const freq = {};
  for (const word of tokens) {
    freq[word] = (freq[word] || 0) + 1;
  }
  const sorted = Object.entries(freq).sort((a, b) => b[1] - a[1]);
  return {
    totalWords: tokens.length,
    uniqueWords: sorted.length,
    topWords: sorted.slice(0, 5).map(([word, count]) => ({ word, freq: count })),
  };
};
